File: class_tool/src/main/java/com/tempsfteam/class_tool/vo/UserVO.java
package com.tempsfteam.class_tool.vo;

import com.tempsfteam.class_tool.constant.role.IdentityEnum;
import com.tempsfteam.class_tool.entity.User;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class UserVO {
    private String name;
    private String avatar;
    private String phone;
    private String email;
    private String userRole;
    private Integer schoolId;
    private Integer classId;

    /**
     * 将 User 转换为 UserVO 的静态方法
     * @param user
     * @return
     */
    public static UserVO convertToUserVO(User user) {
        return new UserVO(
                user.getName(),
                user.getAvatar(),
                user.getPhone(),
                user.getEmail(),
                IdentityEnum.fromValue(user.getRole()).getUserRoleString(),
                user.getSchoolId(),
                user.getClassId()
        );
    }
}

File: class_tool/src/main/resources/static/src/models/vo/userVO.js
export default {
    name: 'userVO',
    title: 'UserVO',
    fields: [
        { name: 'name', label: '用户名', type: 'string' },
        { name: 'avatar', label: '头像', type: 'string' },
        { name: 'phone', label: '电话', type: 'string' },
        { name: 'email', label: '邮箱', type: 'string' },
        { name: 'userRole', label: '角色字符串', type: 'string' },
        { name: 'schoolId', label: '学校ID', type: 'number' },
        { name: 'classId', label: '班级ID', type: 'number' }
    ]
};